bigbite/themer
# [BBT-100] Re-usable typography UI component

## Description



Completes [BBT-100](https://b5ecom.atlassian.net/browse/BBT-100) - This PR creates a re-usable component for typography styles. This is the biggest out of all the styling components, so I've tried to focus on splitting the code where possible.

Where possible I've used the same components that core uses within the global styles/stylebook interface. These are often experimental components with experimental/unstable props, but they are used within core so I don't think its too much of an issue and we can just see what core does if these are changed/removed.

Happy to change this if you think it's wrong, but I've kept the helpers within the `typography` folder and exported them so they can still be used elsewhere. Where I've thought it was necessary (font family and font size), I've created a separate file for the components to avoid huge files that could be difficult to maintain in future.

The only part of this ticket which I've been unable to complete is the `writingMode` property. I'm running 6.3, but when I add `writingMode` to my `theme.json` I can't get it to do anything (within Themer or on the frontend). I've left it for now and added a TODO comment to the top of the file, but will need a new ticket to track it as well.

For the interface, I've used this as a reference but it's not possible to copy this exactly as not all components are shown here: https://www.figma.com/file/fWpNJ5grZmisFBnCUPqrrv/Themer?type=design&node-id=61-2473&mode=dev

## Change Log



- Added typography styles UI component.

## Steps to test



- Add some typography styles to `theme.json` for a specific block so the Typography panel renders
- Ensure the values you entered have been imported correctly
- Ensure the components you _didn't_ enter a value for still render without breaking (they're falling back correctly)
- Play about the values and ensure all the components are working as you expect
- Save & export `theme.json` and ensure the values you had saved are correct

## Screenshots/Videos



### Interface

<img width="654" alt="Screenshot 2023-10-23 at 15 15 40" src="https://github.com/bigbite/themer/assets/45216035/d403e2b0-3a8d-4ec9-a1af-04eee2a638ba">

### Import + changing values

https://github.com/bigbite/themer/assets/45216035/870b64e4-3644-4032-ab37-99f428a7bb96

### Exported theme.json values

<img width="573" alt="Screenshot 2023-10-23 at 15 09 02" src="https://github.com/bigbite/themer/assets/45216035/94792347-5eec-4d51-bf04-ce75f8096c23">

## Checklist:





[BBT-100]: https://b5ecom.atlassian.net/browse/BBT-100?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
